syntax = "proto3";
import "protobufs/Common/Anime.proto";
import "protobufs/Common/User.proto";
import "protobufs/Common/Topic.proto";

message Empty {}

message MostUsedTopics {
    repeated Topic most_used_topics = 1;
}

message Top10_Request{
    string user_name = 1;
}

message Top10_Response{
    repeated Anime animes = 1;
}

message KarmaResponse {
    int32 karma_Value = 1;
}

message GetUserByNameResponse {
    User user = 1;
}

message GetUserByNameRequest {
    string user_name = 1;
}

message GetAllUsersResponse {
    repeated User users = 1;
}

service UserStatisticsService {
    rpc GetMostUsedTopics (Empty) returns (MostUsedTopics); //Prob broken
    rpc GetTop10 (Top10_Request) returns (Top10_Response);
    rpc GetUserKarma (Empty) returns (KarmaResponse);
    rpc GetAllUsers (Empty) returns (GetAllUsersResponse);
    rpc GetUserByName (GetUserByNameRequest) returns (GetUserByNameResponse);
}